docs/tekton-bundle-contracts.md

```diff
@@ -0,0 +1,95 @@
+<!--
+---
+linkTitle: "Tekton Bundle Contracts"
+weight: 8
+---
+-->
+# Tekton Bundle Contract v0.1
+
+When using a Tekton Bundle in a task or pipeline reference, the OCI artifact backing the
+bundle must adhere to the following contract.
+
+## Contract
+
+Only Tekton CRDs (eg, `Task` or `Pipeline`) may reside in a Tekton Bundle used as a Tekton
+bundle reference.
+
+Each layer of the image must map 1:1 with a single Tekton resource.
+
+Each layer must contain all of the following annotations:
+
+- `org.opencontainers.image.title` => `ObjectMeta.Name` of the resource
+- `dev.tekton.image.kind` => `TypeMeta.Kind` of the resource, all lowercased (eg, `task`)
+- `dev.tekton.image.apiVersion` => `TypeMeta.APIVersion` of the resource (eg 
+"tekton.dev/v1alpha1")  
+
+The union of the { `dev.tekton.image.apiVersion`, `dev.tekton.image.kind`, `org.opencontainers.image.title` }
+annotations on a given layer must be unique among all layers of that image. In practical terms, this means no two
+"tasks" can have the same name for example.
+
+Each layer must map 1:1 with a single Tekton resource (eg Task) and have a MIME type of 
+`application/vnd.oci.image.layer.v1.tar+gzip`. Furthermore, each layer must store a tar and gzipped YAML or JSON
```

Comment:
Does it have to be `tar+gzip` or should all the compression be allowed, see https://github.com/opencontainers/image-spec/blob/master/layer.md#image-layer-filesystem-changeset

Reply:
You make a fair point. I think `go-containerregistry` doesn't support zstd but otherwise, from an hour of ref tracing, it should work the same way whether the layer is `tar`, `tar+gzip`, or just `gzip`.

https://github.com/google/go-containerregistry/issues/531